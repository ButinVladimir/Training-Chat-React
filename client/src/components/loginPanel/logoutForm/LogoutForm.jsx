import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

function LogoutForm({ login, onLogout }) {
  return (
    <Fragment>
      <div className="form-row">
        <span>You are loginned as </span>
        <strong>{login}</strong>
      </div>
      <div className="form-row">
        <button type="button" className="logout" onClick={onLogout}>Logout</button>
      </div>
    </Fragment>
  );
}

LogoutForm.propTypes = {
  login: PropTypes.string.isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default LogoutForm;
