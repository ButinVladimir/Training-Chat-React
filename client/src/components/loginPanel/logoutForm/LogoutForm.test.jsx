import React from 'react';
import { shallow } from 'enzyme';
import LogoutForm from './LogoutForm';

describe('LogoutForm', () => {
  it('renders without crashing', () => {
    const component = shallow(<LogoutForm login="login" onLogout={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  it('handles click on logout button', () => {
    const mock = jest.fn();
    const component = shallow(<LogoutForm login="login" onLogout={mock} />);

    component.find('.logout').simulate('click');
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
