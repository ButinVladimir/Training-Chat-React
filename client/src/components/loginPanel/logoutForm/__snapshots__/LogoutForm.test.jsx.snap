// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutForm renders without crashing 1`] = `
<Fragment>
  <div
    className="form-row"
  >
    <span>
      You are loginned as 
    </span>
    <strong>
      login
    </strong>
  </div>
  <div
    className="form-row"
  >
    <button
      className="logout"
      onClick={[Function]}
      type="button"
    >
      Logout
    </button>
  </div>
</Fragment>
`;
